src/puya/client_gen.py

```diff
@@ -74,8 +74,9 @@ def resolve_app_specs(paths: Sequence[Path]) -> Sequence[Path]:
     return app_specs
 
 
-def _parse_app_spec_methods(path: Path) -> tuple[str, Sequence[ARC4Method]]:
-    app_spec = json.loads(path.read_text())
+def parse_app_spec_methods(path_or_str: Path | str) -> tuple[str, Sequence[ARC4Method]]:
+    app_spec_json = path_or_str.read_text() if isinstance(path_or_str, Path) else path_or_str
```

Comment:
`read_text` should be used with `encoding='utf8'` for Windows support.
also, maybe do the `read_text` in the callee to avoid the `isinstance`?